src/tactic/ui/tools/pipeline_wdg.py

```diff
@@ -1568,17 +1569,21 @@ def get_buttons_wdg(my):
 
         project_code = Project.get_project_code()
 
-
-
         button = ButtonNewWdg(title="REFRESH", icon="BS_REFRESH")
         button_row.add(button)
 
         button.add_behavior( {
         'type': 'click_up',
         'cbjs_action': '''
-            var top = bvr.src_el.getParent(".spt_pipeline_editor_top");
-            var top = bvr.src_el.getParent(".spt_pipeline_tool_top");
-            spt.panel.refresh(top);
+            var editor_top = bvr.src_el.getParent(".spt_pipeline_editor_top");
+            if (editor_top && editor_top.hasClass("spt_has_changes")) {
+                if (confirm("Current pipeline has changes.  Do you wish to continue?")) {
```

Comment:
let's use the built in spt.confirm dialog box where u have to define the ok function first.. 
var ok = function() {
     editor_top.removeClass("spt_has_changes");
             spt.panel.refresh(editor_top);

}

spt.confirm("Continue? ", ok);